src/linear_algebra/pi_tensor_product.lean

```diff
@@ -442,6 +444,63 @@ def pempty_equiv : ⨂[R] i : pempty, M ≃ₗ[R] R :=
   map_add' := linear_map.map_add _,
   map_smul' := linear_map.map_smul _, }
 
+section tmul
+
+/-- Collapse a `tensor_product` of `pi_tensor_product`s. -/
+private def tmul : (⨂[R] i : ι, M) ⊗[R] (⨂[R] i : ι₂, M) →ₗ[R] ⨂[R] i : ι ⊕ ι₂, M :=
+tensor_product.lift
+  { to_fun := λ a, pi_tensor_product.lift $ pi_tensor_product.lift
+      (multilinear_map.curry_sum_equiv R _ _ M _ (tprod R)) a,
+    map_add' := λ a b, by simp only [linear_equiv.map_add, linear_map.map_add],
+    map_smul' := λ r a, by simp only [linear_equiv.map_smul, linear_map.map_smul], }
+
+private lemma tmul_apply (a : ι → M) (b : ι₂ → M) :
+  tmul ((⨂ₜ[R] i, a i) ⊗ₜ[R] (⨂ₜ[R] i, b i)) = ⨂ₜ[R] i, sum.elim a b i :=
+begin
+  erw [tensor_product.lift.tmul, pi_tensor_product.lift.tprod, pi_tensor_product.lift.tprod],
+  refl
+end
+
+/-- Expand `pi_tensor_product` into a `tensor_product` of two factors. -/
+private def tmul_symm : ⨂[R] i : ι ⊕ ι₂, M →ₗ[R] (⨂[R] i : ι, M) ⊗[R] (⨂[R] i : ι₂, M) :=
+pi_tensor_product.lift begin
+  apply multilinear_map.dom_coprod,
+  exact tprod R,
+  exact tprod R,
+end
```

Comment:
This looks weird at first glance but I can't find an easier way to solve this. At least a comment will explain why it looks weird.
```suggestion
pi_tensor_product.lift begin -- If we didn't use tactic mode, we'd have to insert a lot of implicit arguments.
  apply multilinear_map.dom_coprod,
  exact tprod R,
  exact tprod R,
end
```

Reply:
```suggestion
-- by using tactic mode, we avoid the need for a lot of `@`s and `_`s
pi_tensor_product.lift $ by apply multilinear_map.dom_coprod; [exact tprod R, exact tprod R]
```